src/test/java/use_cases/GetNews/GetNewsInteractorTest.java

```diff
@@ -0,0 +1,50 @@
+package use_cases.GetNews;
+
+import drivers.Finnhub;
+import org.junit.jupiter.api.Test;
+import use_cases.APIAccessInterface;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class GetNewsInteractorTest {
+
+    @Test
+    void successTest() {
+        GetNewsInputData inputData = new GetNewsInputData("AAPL");
+        APIAccessInterface driverAPI = new Finnhub();
+
+        // This creates a successPresenter that tests whether the test case is as we expect.
+        GetNewsOutputBoundary successPresenter = new GetNewsOutputBoundary() {
+            @Override
+            public void prepareSuccessView(GetNewsOutputData response) {
+                assertEquals("AAPL", response.getTicker());
+
+                List<Map<String, String>> newsItems = response.getNewsItems();
+                assertEquals(5, newsItems.size());
+
+                LocalDateTime now = LocalDateTime.now();
+                LocalDateTime limit = now.minusMonths(1);
+
+                for (Map<String, String> newsItem : newsItems) {
+                    LocalDateTime newsDate = LocalDateTime.parse(newsItem.get("datetime"));
+                    assertTrue(newsDate.isAfter(limit));
+                }
+            }
```

Comment:
Bruh, @rickypramanick, I already removed those interface tests.

Reply:
My bad they hadn't refreshed at the time I started the review

Reply:
> My bad they hadn't refreshed at the time I started the review

Okay, no worries.

Reply:
> My bad they hadn't refreshed at the time I started the review

So can you approve it then because without you it won't work since you requested changes.